src/database-layer/db.ts

```diff
@@ -619,18 +630,37 @@ if (maybePg != null) {
 				}
 			}
 
-			public override async getTxLevelLock(namespaceKey: string, key: number) {
+			public override async getTxLevelLock(
+				namespaceKey: string,
+				key: number,
+				blocking: boolean = true,
+			) {
 				validateTransactionLockParameter(key, 'key');
 				const namespaceId = transactionLockNamespaceMap[namespaceKey];
 				if (namespaceId == null) {
 					throw new Error(
 						`Transaction lock namespace ${namespaceKey} not registered.`,
 					);
 				}
-				await this.executeSql(`SELECT pg_advisory_xact_lock($1, $2);`, [
-					namespaceId,
-					key,
-				]);
+				try {
+					if (blocking) {
+						await this.executeSql(`SELECT pg_advisory_xact_lock($1, $2);`, [
+							namespaceId,
+							key,
+						]);
+						return true;
+					} else {
+						const result = await this.executeSql(
+							`SELECT pg_try_advisory_xact_lock($1, $2);`,
+							[namespaceId, key],
+						);
+						return result.rows[0].pg_try_advisory_xact_lock === true;
+					}
+				} catch (error) {
+					throw new Error(
+						`getTxLevelLock error during getting lock from postgres db layer ${error}`,
```

Comment:
It's minor but we almost always use `err` for these and so the full form of `error` feels kind of out of place
```suggestion
				} catch (err) {
					throw new Error(
						`getTxLevelLock error during getting lock from postgres db layer ${err}`,
```

Reply:
Thanks for the reminder!